src/cryptonote_core/service_node_list.cpp

```diff
@@ -635,4 +638,64 @@ namespace service_nodes
     MERROR("Unable to rollback any further!");
     return false;
   }
+
+  bool convert_registration_args(cryptonote::network_type nettype, const std::vector<std::string>& args, std::vector<cryptonote::account_public_address>& addresses, std::vector<uint32_t>& shares)
+  {
+    if (args.size() % 2 != 0)
+    {
+      MERROR(tr("Expected an even number of arguments: <address> <share> [<address> <share> [...]]"));
+      return false;
+    }
+    addresses.clear();
+    shares.clear();
+    uint64_t total_shares = 0;
+    for (size_t i = 0; i < args.size(); i += 2)
+    {
+      cryptonote::address_parse_info info;
+      if (!cryptonote::get_account_address_from_str(info, nettype, args[i]))
+      {
+        MERROR(tr("failed to parse address"));
+        return false;
+      }
+
+      if (info.has_payment_id)
+      {
+        MERROR(tr("can't use a payment id for staking tx"));
+        return false;
+      }
+      if (info.is_subaddress)
+      {
+        MERROR(tr("can't use a subaddress for staking tx"));
+        return false;
+      }
+
+      addresses.push_back(info.address);
+
+      try
+      {
+        double share_fraction = boost::lexical_cast<double>(args[i+1]);
+        if (share_fraction <= 0 || share_fraction > 1)
+        {
+          MERROR(tr("Invalid share amount: ") << args[i+1] << tr(". ") << tr("Must be more than 0 and no greater than 1"));
+          return false;
+        }
+        uint64_t num_shares = ((uint64_t)STAKING_SHARES) * share_fraction;
+        shares.push_back(num_shares);
+        total_shares += num_shares;
+      }
+      catch (const std::exception &e)
+      {
+        MERROR(tr("Invalid share amount: ") << args[i+1] << tr(". ") << tr("Must be more than 0 and no greater than 1"));
+        return false;
+      }
+    }
+    if (total_shares > (uint64_t)STAKING_SHARES)
+    {
+      MERROR(tr("Invalid share amounts, shares must sum to at most 1."));
+      MERROR(tr("If it looks correct,  this may be because of rounding. Try reducing one of the shareholders shares by a very tiny amount"));
```

Comment:
I think if there's a small margin of error (0.01 * STAKING_SHARES), we should let them know and correct it for them.

Reply:
Nice feature! You should make a Monday task so you don't forget about it

Reply:
Nice delegation.